SU2_CFD/src/solvers/CMeshSolver.cpp

```diff
@@ -593,10 +600,18 @@ void CMeshSolver::ComputeGridVelocity_FromBoundary(CGeometry **geometry, CNumeri
 
   AD::EndPassive(wasActive);
 
+  const su2double velRef = config->GetVelocity_Ref();
+  const su2double invVelRef = 1.0 / velRef;
+
   /*--- Clear residual (loses AD info), we do not want an incremental solution. ---*/
   SU2_OMP_PARALLEL {
     LinSysRes.SetValZero();
-    LinSysSol.SetValZero();
+
+    SU2_OMP_FOR_STAT(omp_chunk_size)
+    for (unsigned long iPoint = 0; iPoint < nPoint; iPoint++)
+      for (unsigned short iDim = 0; iDim < nDim; iDim++)
+        LinSysSol(iPoint, iDim) = geometry[MESH_0]->nodes->GetGridVel(iPoint)[iDim] * velRef;
+    END_SU2_OMP_FOR
```

Comment:
@TobiKattmann for FSI the mesh solver solves two systems, displacement and velocity, these can be initialized to reduced their cost.
This also reduces the cost of clearing indices and recording the primal iteration, in those cases the initialization should be perfect.